sdk/core/azure-core/azure/core/pipeline/transport/_requests_basic.py

```diff
@@ -98,13 +120,19 @@ class StreamDownloadGenerator(object):
 
     :param pipeline: The pipeline object
     :param response: The response object.
+    :keyword bool decompress: If True which is default, will attempt to decode the body based
+        on the ‘content-encoding’ header.
     """
-    def __init__(self, pipeline, response):
+    def __init__(self, pipeline, response, **kwargs):
         self.pipeline = pipeline
         self.request = response.request
         self.response = response
         self.block_size = response.block_size
-        self.iter_content_func = self.response.internal_response.iter_content(self.block_size)
+        decompress = kwargs.get("decompress", True)
```

Comment:
We should `pop` the `decompress` `kwarg` and raise an error if we get unknown `kwarg`s (since we are not pushing them through anywhere else, I believe?)

Reply:
Yes. passed-in kwargs will be thrown away after we read the decompress information.